### ISSUE TITLE
postgresql  deadlock error

### ISSUE THREAD

**Issue by brianmay** [django/channels#905]
This happens when running tests. Not sure I want to spend too much time investigating why this happens until issues like #897 are fixed. Just want to document this for now.

```
[2018-02-14 12:01:32,887: ERROR/django.request] Internal Server Error: /api/internal/projects/
Traceback (most recent call last):
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
psycopg2.extensions.TransactionRollbackError: deadlock detected
LINE 1: ...ion"."codename" FROM "auth_permission" INNER JOIN "accounts_...
                                                             ^
DETAIL:  Process 6613 waits for AccessShareLock on relation 2476770 of database 2475262; blocked by process 6545.
Process 6545 waits for AccessExclusiveLock on relation 2476989 of database 2475262; blocked by process 6613.
HINT:  See server log for query details.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/brian/.pyenv/versions/3.6.4/lib/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/viewsets.py", line 95, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/views.py", line 494, in dispatch
    response = self.handle_exception(exc)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/views.py", line 454, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/views.py", line 491, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/mixins.py", line 45, in list
    return self.get_paginated_response(serializer.data)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/serializers.py", line 742, in data
    ret = super(ListSerializer, self).data
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/serializers.py", line 660, in to_representation
    self.child.to_representation(item) for item in iterable
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/serializers.py", line 660, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/serializers.py", line 504, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/serializers.py", line 504, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/rest_framework/relations.py", line 518, in to_representation
    for value in iterable
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/models/query.py", line 250, in __iter__
    self._fetch_all()
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/models/query.py", line 1118, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 899, in execute_sql
    raise original_exception
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 889, in execute_sql
    cursor.execute(sql, params)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
django.db.utils.OperationalError: deadlock detected
LINE 1: ...ion"."codename" FROM "auth_permission" INNER JOIN "accounts_...
                                                             ^
DETAIL:  Process 6613 waits for AccessShareLock on relation 2476770 of database 2475262; blocked by process 6545.
Process 6545 waits for AccessExclusiveLock on relation 2476989 of database 2475262; blocked by process 6613.
HINT:  See server log for query details.
```

**Comment by andrewgodwin** [comment]
Alright. There's literally nothing I can do about this without more information, so I am inclined to close it (as it's generally my policy that open issues should actually be resolveable). I'll tag it with blocked for now though and will close it once you've had a chance to look further.

**Comment by brianmay** [comment]
Considering all my tests just ran and I don't see any sign of this error, I am going to close this for now. Will reopen if the problem occurs after fix for #897 is merged.
